examples/chabochethermal_example.jl

```diff
@@ -0,0 +1,443 @@
+            push!(stresses, mat.variables.stress[1,1])
+        end
+
+        # cycles
+        eps0 = strains[end]
+        sig0 = stresses[end]
+        temperatures = cycle(Tm, Tb, n)
+        temperature_pairs = zip(temperatures, temperatures[2:end])
+        dstress11 = stress_rate * dt
+        dstress11s = cat(repeat([dstress11], n - 1),
+                         repeat([-dstress11], n - 1),
+                         dims=1)
+        cycle_midpoint = n - 1
+
+        cycle_start_strains = convert(Array{Float64}, [])  # TODO: what's the julianic way to do this?
+        for cycle in 1:ncycles
+            cycle_str = @sprintf("%02d", cycle)
+            println("    cycle $(cycle_str)")
+            push!(cycle_start_strains, strains[end])
+            data_to_print = []
+            for (k, ((Tcurr, Tnext), dstress)) in enumerate(zip(temperature_pairs, dstress11s))
+                if k == 1 || k == cycle_midpoint
+                    push!(data_to_print, (strains[end], stresses[end]))
+                end
+
+                mat.drivers.temperature = Tcurr
+                mat.ddrivers.temperature = Tnext - Tcurr
+                stress_driven_uniaxial_increment!(mat, dstress, dt)
+                update_material!(mat)
+                push!(strains, mat.drivers.strain[1,1])
+                push!(stresses, mat.variables.stress[1,1])
+            end
+
+            strains_to_print, stresses_to_print = (collect(col) for col in zip(data_to_print...))
+            strains_to_print = format_numbers(strains_to_print)
+            stresses_to_print = format_numbers(stresses_to_print)
+            println("        start    ε11 = $(strains_to_print[1]), σ11 = $(stresses_to_print[1])")
+            println("        midpoint ε11 = $(strains_to_print[2]), σ11 = $(stresses_to_print[2])")
+        end
+
+        println("Strain at cycle start:")
+        cycle_start_strains_to_print = format_numbers(cycle_start_strains)
+        diffs = diff(cycle_start_strains)
+        diffs_to_print = cat([nothing], format_numbers(diffs), dims=1)
+        for (cycle, (strain, dstrain)) in enumerate(zip(cycle_start_strains_to_print, diffs))
+            cycle_str = @sprintf("%02d", cycle)
+            println("    cycle $(cycle_str), ε11 = $(strain), Δε11 w.r.t. previous cycle = $(dstrain)")
+        end
+
+        p4 = plot(strains, stresses, label="\$\\sigma(\\varepsilon)\$")
+        scatter!([eps0], [sig0], markercolor=:blue, label="First cycle start")
+        xlabel!("\$\\varepsilon\$")
+        ylabel!("\$\\sigma\$")
+        title!("Non-symmetric stress cycle, $(ncycles) cycles")
+
+        # --------------------------------------------------------------------------------
+        # plot the results
+
+        # https://docs.juliaplots.org/latest/layouts/
+        plot(p1, p2, p3, p4, layout=(2, 2))
+
+        # TODO:
+        # - more tests based on Bari's thesis?
+        # - use Abaqus as reference point? Ask Joona.
```

Comment:
This is a good idea. I will look into providing a reference result.

Reply:
Ok. Thanks! This will help in writing actual unit tests.

Reply:
I sent you via email the results (time, strains, stresses, temperatures) for this anisothermal test and the material specifications used for the test:
![image](https://user-images.githubusercontent.com/1643320/110073605-4fa26880-7d88-11eb-8f88-c57b9dbade5c.png)